client/src/main/java/se/fortnox/reactivewizard/client/HttpClient.java

```diff
@@ -509,6 +519,16 @@ protected String getPath(Method method, Object[] arguments, JaxRsMeta meta) {
                     } else {
                         path = path.replaceAll("\\{" + ((PathParam)annotation).value() + "\\}", this.urlEncode(this.serialize(value)));
                     }
+                } else if (annotation instanceof BeanParam) {
```

Comment:
It might be a bit complex to mutate the list we are iterating over?

Would it be possible to do the "conversion of bean params" to normal params earlier, to make the code easier to understand? So that we break down the process into steps, instead of doing everything at once?

Reply:
Doing the conversion of BeanParams first would require to iterate the arguments twice. From my point of view, this implementation is simple, it iterates each argument and deals with each case one by one.

I plan to rewrite HttpClient in such a way that when the client is created, all reflection is done and at runtime we have a set of functions that will deal with the known arguments without a single if-statement. But that is not something I wanted to do in the same PR as adding this functionality.